src/interface.jl

```diff
@@ -0,0 +1,655 @@
+
+inv(ct::Composed) = Composed(map(inv, reverse(ct.ts)))
+
+# # TODO: should arrays also be using recursive implementation instead?
+function (cb::Composed{<: AbstractArray{<: Bijector}})(x)
+    res = x
+    for b ∈ cb.ts
+        res = b(res)
+    end
+
+    return res
+end
+
+# recursive implementation like this allows type-inference
+_transform(x, b1::Bijector, b2::Bijector) = b2(b1(x))
+_transform(x, b::Bijector, bs::Bijector...) = _transform(b(x), bs...)
+(cb::Composed{<: Tuple})(x) = _transform(x, cb.ts...)
+
+function _logabsdetjac(x, b1::Bijector, b2::Bijector)
+    res = forward(b1, x)
+    return logabsdetjac(b2, res.rv) .+ res.logabsdetjac
+end
+function _logabsdetjac(x, b1::Bijector, bs::Bijector...)
+    res = forward(b1, x)
+    return _logabsdetjac(res.rv, bs...) .+ res.logabsdetjac
+end
+logabsdetjac(cb::Composed, x) = _logabsdetjac(x, cb.ts...)
+
+# recursive implementation of `forward`
+# HACK: we need this one in the case where `length(cb.ts) == 2`
+# in which case forward(...) immediately calls `_forward(::NamedTuple, b::Bijector)`
+function _forward(f::NamedTuple, b::Bijector)
+    y, logjac = forward(b, f.rv)
+    return (rv=y, logabsdetjac=logjac .+ f.logabsdetjac)
+end
+function _forward(f::NamedTuple, b1::Bijector, b2::Bijector)
+    f1 = forward(b1, f.rv)
+    f2 = forward(b2, f1.rv)
+    return (rv=f2.rv, logabsdetjac=f2.logabsdetjac .+ f1.logabsdetjac .+ f.logabsdetjac)
+end
+function _forward(f::NamedTuple, b::Bijector, bs::Bijector...)
+    f1 = forward(b, f.rv)
+    f_ = (rv=f1.rv, logabsdetjac=f1.logabsdetjac .+ f.logabsdetjac)
+    return _forward(f_, bs...)
+end
+_forward(x, b::Bijector, bs::Bijector...) = _forward(forward(b, x), bs...)
+forward(cb::Composed{<: Tuple}, x) = _forward(x, cb.ts...)
+
+function forward(cb::Composed, x)
+    rv, logjac = forward(cb.ts[1], x)
+    
+    for t in cb.ts[2:end]
+        res = forward(t, rv)
+        rv = res.rv
+        logjac = res.logabsdetjac .+ logjac
+    end
+    return (rv=rv, logabsdetjac=logjac)
+end
+
+##############################
+# Example bijector: Identity #
+##############################
+
+struct Identity <: Bijector end
+(::Identity)(x) = x
+(::Inversed{Identity})(y) = y
+
+forward(::Identity, x) = (rv=x, logabsdetjac=zero(x))
+
+logabsdetjac(::Identity, y::T) where T <: Real = zero(T)
+logabsdetjac(::Identity, y::AbstractArray{T}) where T <: Real = zero(T)
+
+const IdentityBijector = Identity()
+
+###############################
+# Example: Logit and Logistic #
+###############################
+using StatsFuns: logit, logistic
+
+struct Logit{T<:Real} <: Bijector
+    a::T
+    b::T
+end
+
+(b::Logit)(x) = @. logit((x - b.a) / (b.b - b.a))
+(ib::Inversed{<: Logit{<: Real}})(y) = @. (ib.orig.b - ib.orig.a) * logistic(y) + ib.orig.a
+
+logabsdetjac(b::Logit{<:Real}, x) = @. log((x - b.a) * (b.b - x) / (b.b - b.a))
+
+#############
+# Exp & Log #
+#############
+
+struct Exp <: Bijector end
+struct Log <: Bijector end
+const exp_b = Exp()
```

Comment:
Why do we need this?

Reply:
Why do we need `exp_b`? Maybe we don't. I just figured that these are stateless and so we could just use a single instance which we plug everywhere instead of constructing `n` different `Exp` or whatever.

I suspect the Julia compiler will just do the most efficient thing for us anyway, so might very well be redundant, but I figured I'd be on the safe side.

Remove and just use `Exp()` instead?

Reply:
Yes, I think that's more readble. 

When things are imutable plugging the constructor is fine.

```julia
julia> struct A end

julia> a1 = A()
A()

julia> a2 = A()
A()

julia> a1 == a2
true

julia> mutable struct B end

julia> b1 = B()
B()

julia> b2 = B()
B()

julia> b1 == b2
false
```

Reply:
Sounds good to me :+1: I'll get it fixed asap